### PR TITLE
sink(ticdc): use  an unbounded chan in conflict detector

### LINKED ISSUE #8255: use  an unbounded chan in conflict detector
use  an unbounded chan in conflict detector

### PR DESCRIPTION
**Pull request by sdojjy**
### What problem does this PR solve?

Issue Number: close #8255

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)

### Release note 

```release-note
`None`.
```

### CODE REVIEW

File: pkg/causality/conflict_detector.go [hunk 1]
```diff
@@ -37,8 +37,8 @@ type ConflictDetector[Worker worker[Txn], Txn txnEvent] struct {
 	nextWorkerID atomic.Int64
 
 	// Used to run a background goroutine to GC or notify nodes.
-	notifiedNodes *containers.SliceQueue[func()]
-	garbageNodes  *containers.SliceQueue[txnFinishedEvent]
+	notifiedNodes *chann.Chann[func()]
```
[the remark this answers is not in the document]
Comment (sdojjy) [reply pkg/causality/conflict_detector.go:40]: done

**Review by hicqu — APPROVED** [review]

**Review by nongfushanquan — APPROVED** [review]

**Pull request merged**